kratos/solving_strategies/strategies/solving_strategy.h

```diff
@@ -29,426 +29,493 @@
-    typedef typename DofsArrayType::const_iterator DofConstantIterator;
-    /*@} */
+    
+    ///@}
+    ///@name Life Cycle
+    ///@{
 
     /** Constructor.
+     * @param rModelPart: The model part to be computed
+     * @param MoveMeshFlag: The flag to set if the mesh is moved or not
      */
 
-    /*@{ */
-
     SolvingStrategy(
-        ModelPart& model_part, bool MoveMeshFlag = false
+        ModelPart& rModelPart, 
+        bool MoveMeshFlag = false
     )
-        : mr_model_part(model_part)
+        : mrModelPart(rModelPart)
     {
         SetMoveMeshFlag(MoveMeshFlag);
     }
-    /*@} */
+    
+    ///@}
 
     /** Destructor.
      */
 
-    /*@{ */
-    virtual ~SolvingStrategy()
-    {
-    }
-    /*@} */
+    ///@{
+    virtual ~SolvingStrategy(){}
+    
+    ///@}
 
-    //*********************************************************************************
     /**OPERATIONS ACCESSIBLE FROM THE INPUT:*/
-    /*@{ */
+    
+    ///@{
 
     /**
-    operation to predict the solution ... if it is not called a trivial predictor is used in which the
-    values of the solution step of interest are assumed equal to the old values
+     * Operation to predict the solution ... if it is not called a trivial predictor is used in which the
+     * values of the solution step of interest are assumed equal to the old values
      */
     virtual void Predict()
     {
+        KRATOS_ERROR << "You are calling to the base class method Predict, you are evil, and your seed must be eradicated from the face of the earth" << std::endl;
     }
 
     /**
-    Initialization of member variables and prior operations
+     * Initialization of member variables and prior operations
      */
     virtual void Initialize()
     {
+        KRATOS_ERROR << "You are calling to the base class method Initialize, you are evil, and your seed must be eradicated from the face of the earth" << std::endl;
     }
 
     /**
-    the problem of interest is solved.
-	This function calls sequentially: Initialize(), InitializeSolutionStep(), Predict(), SolveSolutionStep() and FinalizeSolutionStep().
-	All those functions can otherwise be called separately.
+     * The problem of interest is solved.
+     * This function calls sequentially: Initialize(), InitializeSolutionStep(), Predict(), SolveSolutionStep() and FinalizeSolutionStep().
+     * All those functions can otherwise be called separately.
      */
     virtual double Solve()
     {
-		Initialize();
-		InitializeSolutionStep();
-		Predict();
-		SolveSolutionStep();
-		FinalizeSolutionStep();
-        return 0.00;
+        Initialize();
+        InitializeSolutionStep();
+        Predict();
+        SolveSolutionStep();
+//         const bool IsConverged = SolveSolutionStep();
+        FinalizeSolutionStep();
+        
+        // NOTE: Here I would add a conversion to double of the boolean IsConverged, that way we can know if the problem has is converged
```

Comment:
@RiccardoRossi @pooyan-dadvand I added this as suggestion, right now is commented, but maybe is a good idea to use the double output from the Solve to know if the problem converged (I know that this is done in the convergence criteria, but This can be use for problem with zero verbosity, or checks)